extensions/wikia/AdEngine/AdEngine2ContextService.class.php

```diff
@@ -17,8 +17,11 @@ public function getContext( Title $title, $skinName ) {
 			$adPageTypeService = new AdEngine2PageTypeService();
 			$wikiaPageType = new WikiaPageType();
 			$pageType = $wikiaPageType->getPageType();
-			$hasFeaturedVideo = !empty( $wg->EnableArticleFeaturedVideo )
-				&& ArticleVideoContext::isFeaturedVideoEmbedded( $title->getArticleID() );
+			$articleId = $title->getArticleID();
+			$hasFeaturedVideo = ( !empty( $wg->EnableArticleFeaturedVideo )
+				&& ArticleVideoContext::isFeaturedVideoEmbedded( $articleId ) ) ||
+				// XW-4713 | Hide UAPs on pages running the Recommended Video ABTest
+				ArticleVideoContext::isRecommendedVideoAvailable( $articleId );
 			// pages with featured video on mercury have no ATF slots
 			$delayBtf = ( $skinName === 'mercury' && $hasFeaturedVideo ) ? false : $wg->AdDriverDelayBelowTheFold;
```

Comment:
@Wikia/adeng What does this exactly do? Can we be harmed by setting it to true on recommended video pages? (there will be only around 20 of these pages for the test).